### PR TITLE
Support for gzip middleware

### PR DESCRIPTION
**Pull request by Josca**
Please merge this improvement or bug fix into this package.

gin-swagger does not work correctly when using gzip middleware from <https://github.com/gin-contrib/gzip>. In this case response header `Content-Type` was `application/x-gzip` for all returned entities (index.html, doc.json, etc.). So browser offered file downloading instead of rendering swagger html page.

It was already discussed in [swaggo issue](https://github.com/swaggo/swag/issues/174). So I solved this by explicitly adding response headers.

I didn't find better solution then explicitly write `Content-Type` response header according to returned item. See it [here](https://github.com/Josca/gin-swagger/blob/61d00f5f41e8a189c31ebb30b6b6a3fa831486dd/swagger.go#L65).

This solution works in both cases - with and without gzip middleware. I added example and also test for it.

### CODE REVIEW

**Comment by pei0804** [comment]
@easonlin404 Can you review?

File: swagger.go [hunk 1]
```diff
@@ -61,6 +62,16 @@ func CustomWrapHandler(config *Config, h *webdav.Handler) gin.HandlerFunc {
...
 
+		if strings.HasSuffix(path, ".html") {
+			c.Header("Content-Type", "text/html; charset=utf-8")
+		} else if strings.HasSuffix(path, ".css") {
+			c.Header("Content-Type", "text/css; charset=utf-8")
+		} else if strings.HasSuffix(path, ".js") {
+			c.Header("Content-Type", "application/javascript")
+		} else if strings.HasSuffix(path, ".json") {
+			c.Header("Content-Type", "application/json")
+		}
```
Comment (easonlin404) [inline swagger.go]: This part seems like can have better solutions.  It can be improved for the new PR.

**Comment by easonlin404** [comment]
@Josca would you help add examples at README?

File: swagger.go (hunk 1 above)
Comment (Josca) [reply swagger.go]: Yes, as I said it's not the best solution, but it works. I don't know why header was overwritten with `application/x-gzip` value before. This only fixes consequences, not origins.

**Comment by Josca** [comment]
> @Josca would you help add examples at README?

I'm not sure it's needed. It only enables this package usage in new usecase. It doesn't add new feature.

**Pull request merged**